src/main/java/org/tugraz/sysds/runtime/lineage/LineageItem.java

```diff
@@ -0,0 +1,93 @@
+package org.tugraz.sysds.runtime.lineage;
+
+import java.util.ArrayList;
+
+import org.tugraz.sysds.runtime.instructions.cp.CPOperand;
+
+public class LineageItem {
+    private static int _id_counter = 1;
```

Comment:
Please an IDSequence, which ensures atomic updates in multi-threaded environments, uses long (2G of int would easily overflow).